enterprise/backend/test/metabase_enterprise/serialization/v2/e2e_test.clj

```diff
@@ -735,3 +735,56 @@
                                           (cmd/v2-load! dump-dir {}))
                         "throws")))))))))))
+
+(deftest pivot-export-test
+  (testing "Pivot table export and load correctly"
+    (let [card1s (atom nil)]
+      (ts/with-random-dump-dir [dump-dir "serdesv2-"]
+        (ts/with-source-and-dest-dbs
+          (ts/with-source-db
+            (mt/dataset sample-dataset
+              (mt/$ids nil
+                (t2.with-temp/with-temp
+                  [Database   {db-id :id}    {:name "Pivot Database"}
+                   Collection {coll-id :id}  {:name "Pivot Collection"}
+                   Card       card           {:name          "Pivot Card"
+                                              :collection_id coll-id
+                                              :dataset_query {:type     :query
+                                                              :database db-id
+                                                              :query    {:source-table $$orders
+                                                                         :aggregation  [:sum [:field %orders.id nil]]
+                                                                         :breakout     [[:field %orders.user_id nil]]}}
+                                              :visualization_settings
+                                              {:pivot_table.column_split
+                                               {:rows    [[:field %people.name {:base-type    :type/Text
+                                                                                :source-field %orders.user_id}]]
+                                                :columns [[:field %products.title {:base-type    :type/Text
+                                                                                   :source-field %orders.product_id}]]
+                                                :values  [[:aggregation 0]]}
+                                               :column_settings
+                                               {(format "[\"ref\",[\"field\",%s,null]]" %people.name)
+                                                {:pivot_table.column_sort_order "descending"}}}}]
+                  (reset! card1s card)
+                  (-> (serdes/with-cache (into [] (extract/extract {})))
+                      (storage/store! dump-dir))))))
+
+          (ts/with-dest-db
+            (mt/dataset sample-dataset
+              (mt/$ids nil
+                (serdes/with-cache (serdes.load/load-metabase! (ingest/ingest-yaml dump-dir)))
+
+                (let [viz (t2/select-one-fn :visualization_settings Card :entity_id (:entity_id @card1s))]
+                  (is (contains? viz :pivot_table.column_split))
+                  ;; column ids are different in different dbs
+                  (is (not= (get-in @card1s [:visualization_settings :pivot_table.column_split :rows 0 1])
+                            (get-in viz [:pivot_table.column_split :rows 0 1])))
```

Comment:
@calherries I'm a bit afraid this line will flake (there is no promise that ids will indeed be different), but it's like an insurance that we're getting some real action here.

Reply:
Yeah, that's not ideal. Can you find a way to ensure that the test IDs are different, at least with high probability? I'm thinking of a hack like creating an extra database in the destination DB with one field and one table.

Reply:
Maybe you can use `mt/with-temp` for this. Just create a database, table and a field before calling `mt/dataset` to be safe

Reply:
Results are not reassuring. :) So if I leave test in its current form, ids are 21 (source) vs 7 (dest). If I create db+table+field, it because 21 vs 24, which is closer and makes me uneasy. :\ I'm thinking of living it in place with some comment that if it flakes, here's the discussion.